ui/src/app/base/types.ts

```diff
@@ -1,8 +1,14 @@
 export type TSFixMe = any; // eslint-disable-line @typescript-eslint/no-explicit-any
 
-export type Sort = {
-  direction: "ascending" | "descending" | "none";
-  key: string;
+export enum SortDirection {
+  ASCENDING = "ascending",
+  DESCENDING = "descending",
+  NONE = "none",
+}
```

Comment:
I may be missing some context here, what was your reasoning for choosing `enum`? Have you considered using an object with `as const`?

https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums

```suggestion
export const SortDirection = {
  ASCENDING: "ascending",
  DESCENDING: "descending",
  NONE: "none",
} as const
```

Reply:
I'm personally a fan of this method also - we've started using it in our [react-components library](https://github.com/canonical-web-and-design/react-components/blob/master/src/components/Notification/Notification.tsx#L10) in conjunction with a helper type [ValueOf](https://github.com/canonical-web-and-design/react-components/blob/master/src/types/index.ts#L41) (`typeof Foo[keyof typeof Foo]` is a bit of a mouthful).

Still, one of the benefits (and caveats) of using `enum`s like this though is that you *have* to use them - you can't just use the object values directly. So then there's literally only one place you ever have to write the underlying value.

Reply:
@petermakowski this is a great question and like @Caleb-Ellis said we've started to adopt that pattern in some places.

I think we need to investigate it a bit more as it hasn't always worked brilliantly (i.e. having to add `as const` when you're using strings in some places).

I've updated this to use `as const`.